### PR TITLE
Upgrade: eslint-plugin-node to 7.0.1

### PR DESCRIPTION
**Pull request by mysticatea**
**What is the purpose of this pull request? (put an "X" next to item)**

[X] Other, please explain: upgrade `eslint-plugin-node` to `7.0.1` for linting our codebase.

**What changes did you make? (Give an overview)**

It has new features:

- It reports the APIs of Node.js built-in modules and globals if those are not supported on the configured Node.js versions (`^6.14.0 || ^8.10.0 || >=9.10.0`).
- It reports newly deprecated APIs.

**Is there anything you'd like reviewers to focus on?**

I'm not sure if this `util.isDeepStrictEqual()` part is needed.
That looks to do double checking for the AST if `util.isDeepStrictEqual` exists. But why?

```js
// Feature detect the Node.js implementation and use that if available.
if ((util.isDeepStrictEqual && !util.isDeepStrictEqual(beforeAST, afterAST)) || !lodash.isEqual(beforeAST, afterAST)) {
```

### CODE REVIEW

**Comment by Aladdin-ADD** [comment]
> I'm not sure if this util.isDeepStrictEqual() part is needed.
> That looks to do double checking for the AST if util.isDeepStrictEqual exists. But why?

seems we have been inconsistent in the usage of `_. isEqual ` vs.`util.isDeepStrictEqual`. it can be restricted in the eslintrc( but in another PR)

File: lib/testers/rule-tester.js [hunk 1]
```diff
@@ -398,6 +398,7 @@ class RuleTester {
         function assertASTDidntChange(beforeAST, afterAST) {
 
             // Feature detect the Node.js implementation and use that if available.
+            // eslint-disable-next-line node/no-unsupported-features/node-builtins
```
Comment (platinumazure) [inline lib/testers/rule-tester.js]: I'm a bit confused about this line. Is the name of the rule `"no-unsupported-features/node-builtins"` (and therefore the plugin-qualified name is `"node/no-unsupported-features/node-builtins"`? I don't think I've ever seen a rule defined with a forward slash in its name, just wanted to make sure I understood correctly.

Comment (mysticatea) [reply lib/testers/rule-tester.js]: Yes. We can use `/` in the rule name in plugins. https://github.com/eslint/eslint/issues/3458#issuecomment-257161846 mentioned this fact.

File: lib/testers/rule-tester.js [hunk 2]
```diff
@@ -398,6 +398,7 @@ class RuleTester {
...
             if ((util.isDeepStrictEqual && !util.isDeepStrictEqual(beforeAST, afterAST)) || !lodash.isEqual(beforeAST, afterAST)) {
```
[the remark this answers is not in the document]
Comment (mysticatea) [reply lib/testers/rule-tester.js]: Yes, it's exactly a code I thought.
But I'm wondering we can rewrite it to simple `if (lodash.isEqual(beforeAST, afterAST)) {`.
This feature detection might cause to change test results by the existence of `util.isDeepStrictEqual`.

**Comment by platinumazure** [comment]
I think the thought was that `util.isDeepStrictEqual` might be faster, but
I have no objection to just using lodash.

**Comment by Aladdin-ADD** [comment]
`util.isDeepStrictEqual` was introduced in Node.js 9.0.0 (https://nodejs.org/api/util.html#util_util_isdeepstrictequal_val1_val2)

**Comment by mysticatea** [comment]
@Aladdin-ADD It's the reason `util.isDeepStrictEqual` was warned by the `node/no-unsupported-features/node-builtins` rule.

Problem is:

```js
!util.isDeepStrictEqual(beforeAST, afterAST) || !lodash.isEqual(beforeAST, afterAST)
```

That `||` is definitely nonsense. So I hope to fix it, but there are two ways:

1. Ternary operator:

    ```js
    !(util.isDeepStrictEqual ? util.isDeepStrictEqual(beforeAST, afterAST) : lodash.isEqual(beforeAST, afterAST))
    ```

2. Remove the `util.isDeepStrictEqual`:

    ```js
    !lodash.isEqual(beforeAST, afterAST)
    ```

I prefer the latter because we don't need the forking by feature detection.

**Comment by mysticatea** [comment]
I sent a PR: #10617

**Comment by Aladdin-ADD** [comment]
might need to rebase, since #10617 get merged. cc @mysticatea

**Comment by mysticatea** [comment]
Done!

**Pull request merged**